### PR TITLE
Do not allow to specify both image and dockerfile in configuration.

### PR DESCRIPTION
**Pull request by mrfuxi**
Config validation mentioned in #1908

Signed-off-by: Karol Duleba <mr.fuxi@gmail.com>

### CODE REVIEW

**Comment by dnephin** [comment]
LGTM

Build failures are known (#1922) not related to your change.

A little more context could be included about why build/dockerfile is incompatible with image, but I  don't think that's necessary to merge this PR.

**Comment by mrfuxi** [comment]
We could quote exact reason form error message: 

> Using `image` together with either `build` or `dockerfile` is not allowed. Attempting to do so results in an error. A service can either be built to image or use an existing image, not both.

File: tests/unit/config_test.py [hunk 1]
```diff
@@ -191,6 +191,28 @@ def test_invalid_list_of_strings_format(self):
                 )
             )
 
+    def test_config_image_and_build_raise_validation_error(self):
```
Comment (mnowster) [inline tests/unit/config_test.py]: This is a duplicate test, we already have the image and build case covered here: https://github.com/mrfuxi/compose/blob/image-dockerfile-validation/tests/unit/config_test.py#L131

**Comment by mnowster** [comment]
Once the duplicate test has been removed, everything else is LGTM. 

Thanks @mrfuxi :star:

**Comment by mrfuxi** [comment]
Ok, will remove the test (have not noticed it earlier). Should I update docs or leave them as they are?

**Comment by dnephin** [comment]
I think they're ok as they are

**Comment by mrfuxi** [comment]
Extra test if gone

**Pull request merged**